pkg/lib/serverTime.js

```diff
@@ -321,13 +319,178 @@
+                        if (line == "# Added by Cockpit")
+                            in_cockpit_section = true;
+                        else {
+                            const fields = line.split(" ");
+                            if (fields[0] == "server") {
+                                if (in_cockpit_section)
+                                    result.extra.push(fields[1]);
+                                else
+                                    result.standard.push({ name: fields[1], enabled: true });
+                            } else {
+                                in_cockpit_section = false;
+                                if (fields[0] == "pool")
+                                    result.standard.push({ name: fields[1], enabled: true });
+                                if (fields[0] == "#pool" || fields[0] == "#server")
+                                    result.standard.push({ name: fields[1], enabled: false });
+                                if (fields[0] == "sourcedir")
+                                    result.sources.push({ name: fields[1], enabled: true });
+                                if (fields[0] == "#sourcedir")
+                                    result.sources.push({ name: fields[1], enabled: false });
+                            }
+                        }
+                    }
+
+                    return result;
+                });
+    }
+
+    function set_ntp_config_chronyd(config) {
+        const chrony_file = cockpit.file(config.file, { superuser: "try" });
+
+        function confline(line, list, name) {
+            const c = list.find(e => e.name == name);
+            if (!c)
+                return line;
+            const line_trim = line.trim();
+            const line_enabled = !line_trim.startsWith("#");
+            if (c.enabled && !line_enabled)
+                return line_trim.substr(1);
+            if (!c.enabled && line_enabled)
+                return "#" + line_trim;
+            return line;
+        }
+
+        return chrony_file.modify(text => {
+            let new_text = "";
+
+            let in_cockpit_section = false;
+            for (const line of text.trimEnd().split("\n")) {
+                if (line == "# Added by Cockpit")
+                    in_cockpit_section = true;
+                else {
+                    const fields = line.split(" ");
+                    const was_in_cockpit_section = in_cockpit_section;
+                    if (fields[0] != "server")
+                        in_cockpit_section = false;
+                    if (!in_cockpit_section && (!was_in_cockpit_section || line != "")) {
+                        if (fields[0] == "pool" || fields[0] == "#pool")
+                            new_text += confline(line, config.standard, fields[1]) + "\n";
+                        else if (fields[0] == "server" || fields[0] == "#server")
+                            new_text += confline(line, config.standard, fields[1]) + "\n";
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17992-20230119-141051-b2cd92d7-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)